Adding README link to Reusing Swagger Parameters wiki page

nburkley opened the pull request:
I've added a wiki page with some details on how to reuse parameters across actions/controllers, following @mbuhot's great help with #115. 

[Reusing swagger parameters](https://github.com/xerions/phoenix_swagger/wiki/Reusing-swagger-parameters)

I thought it might be best to put this in the Wiki and link to it from the README so it doesn't clutter things up too much.

mbuhot commented:
This is great, thanks @nburkley!

@surik: In general do you want to keep additional documentation like this in the github wiki, or should we publish it to hexdoc.pm similar to the new [phoenix guides](https://hexdocs.pm/phoenix/overview.html)?

surik commented:
@mbuhot I would do it on hexdocs.pm.

mbuhot commented:
@nburkley Do you want to try copying the wiki page contents to a markdown file (maybe in a new /docs) directory and updating the `mix.exs` file to include that file in the docs?

Looks like you can just include it in the `extras` list along with `README.md`:

```elixir
  docs: [extras: ["README.md"], main: "readme", source_ref: "v#{@version}"]]
```

nburkley commented:
@mbuhot Sure, I'll it with me.

nburkley commented on README.md:
```diff
@@ -137,6 +137,10 @@ swagger_path :index do
 
+The `swagger_path` macro layer is just some syntactic sugar over regular elixir functions. Therefore it can easily be extended, for instance, if we want to reuse some common parameters.
+
+For more details on this take a look at [Reusing Swagger Parameters](https://hexdocs.pm/phoenix_swagger/reusing-swagger-parameters.html).
```
I ran `mix docs` locally and this link _should_ be correct once it's added to hexdocs.

mbuhot replied:
Looks like you can just use a relative link: 

```markdown
[Reusing Swagger Parameters](reusing-swagger-parameters.html)
```

I noticed this is how the phoenix guides link between each other:

```markdown
For now, we'll ignore the pipelines and the use of `scope` here and just focus on adding a route. (We cover these topics in the [Routing Guide](routing.html), if you're curious.)
```

mbuhot replied:
Oops, that will work within Hexdocs but will 404 if clicked from the Github readme.
Lets keep the full URL. 👍

nburkley commented:
@mbuhot I've deleted the wiki page and moved the contents into a markdown file. 

It looks good locally:
<img width="650" alt="screen shot 2017-07-31 at 22 10 40" src="https://user-images.githubusercontent.com/85859/28796291-47ee303e-763d-11e7-8fa5-acb920c2093e.png">

Let me know if there's anything else.

mbuhot commented:
@surik you might be able to republish the docs with `mix hex.publish docs`.
I'm not sure if it requires a new package version?

surik commented:
Published to https://hexdocs.pm/phoenix_swagger/0.6.4
